hwdetect/hwdetect.sh

```diff
@@ -1,68 +1,57 @@
 #!/bin/bash
 
 #TODO: Detect plugged and unplugged irt
-#	
 
 
 
 
-#somewhy this function gets an error ./hwdetect.sh: line 15: i2c_devices: command not found
-
-#function plugged_i2c()
-#{
-#	i2c_devices = $(ls /dev/i2c-*)
-#	echo "****Connected I2C****"
-#	for i2c in $i2c_devices
-#	do
-#		echo $i2c
-#	done
-#}
+function plugged_i2c()
+{
+	i2c_devices=$(ls /dev/i2c-*)
+	if [ -n "$i2c_devices" ] ; then
+		echo "****Connected I2C****"
+		for i2c in $i2c_devices ; do
+			echo $i2c
+		done
+	fi
+}
 
 #credits to olegovich22, like his implementation
 function plugged_flash_drivers()
 {
 	flash_devices=$(ls /dev/sd*)
-	if [ -n "$flash_devices" ]
-	then
-
-	echo
-	echo "****Connected Flash drivers****"
-
-	for d in $flash_devices 
-	do
-		tput setaf 1
-	 	echo -n "Detected flash driver controler on "
-		echo $d
-		tput sgr0
-		tput setaf 2
-		echo $(udevadm info $d | grep -oE "MODEL=.*")
-		echo $(udevadm info $d | grep -oE "VENDOR=.*")
-		echo -n $(udevadm info $d | grep -oE "BUS=.*")
-		tput sgr0
+	if [ -n "$flash_devices" ] ; then 
 		echo
-	done
-
-	echo
-
+		echo "****Connected Flash drivers****"
+
+		for d in $flash_devices ; do
+			tput setaf 1
+		 	echo -n "Detected flash driver controler on "
+			echo $d
+			tput sgr0
+			tput setaf 2
+			echo $(udevadm info $d | grep -oE "MODEL=.*")
+			echo $(udevadm info $d | grep -oE "VENDOR=.*")
+			echo -n $(udevadm info $d | grep -oE "BUS=.*")
+			tput sgr0
+			echo
+		done
 	fi
 }
 
 
-echo "Checking connected HW"
-
+echo "****Checking connected HW****"
+echo " "
 
-echo "****Connected I2C****"
-for i2c in $(ls /dev/i2c-*)
-do
-	echo $i2c
-done
 
+plugged_i2c
 plugged_flash_drivers
 
+
 while true
 do
 	echo "Detecting plugged and unplugged hardware. TBD"
```

Comment:
Are you going to continue working on this script?

Reply:
I have made changes to the current version and functionality (i2c + flash and no real time detection, commit d0a1332). If they are ok, I would suggest to merge this patch for now. 
I will come back and extend functionality when I have more time, but I see it as another patch. 
If you insist on working more on this topic, I will change my priorities.

Reply:
I'll wait.